src/engine/IndexScan.h

```diff
@@ -11,13 +11,19 @@ class SparqlTriple;
 class SparqlTripleSimple;
 
 class IndexScan final : public Operation {
+  using Graphs = ScanSpecificationAsTripleComponent::Graphs;
```

Comment:
1. An index scan with added prefilters can have a changed (more accurate) size estimate. 
2. The exactSize() function (probably not important for the case here, but still broken) also changes with block prefilters
3. (VERY IMPORTANT) The cache key of a prefiltered index scan is changed. change `getCacheKeyImpl` and also test these changes.